measurements/pages/__init__.py

```diff
@@ -204,34 +208,84 @@ def files_in_country(country_code):
                            order_by=order_by,
                            current_country=country_code)
 
+def decompress_autoclaved(
+        autoclaved_filename,
+        frame_off,
+        total_frame_size,
+        intra_off,
+        report_size
+    ):
+    url = urljoin(current_app.config['AUTOCLAVED_BASE_URL'], autoclaved_filename)
+    def generator():
+        try:
+            headers = {"Range": "bytes={}-{}".format(frame_off, frame_off + total_frame_size)}
+            r = requests.get(url, headers=headers, stream=True)
+            streamed_data = 0
+            while streamed_data < report_size:
+                for chunk in lz4framed.Decompressor(r.raw):
+                    # I am at the beginning of the stream and I need to send some
+                    # offset into the data
+                    if streamed_data == 0 and intra_off > 0:
+                        chunk = chunk[intra_off:]
+                    # Avoid streaming more data than I should
+                    if (streamed_data + len(chunk)) > report_size:
+                        chunk = chunk[:abs(streamed_data - report_size)]
+                    yield chunk
+                    streamed_data += len(chunk)
```

Comment:
Do you want to add sanity-check here that the very first byte of the streamed data is `{` and the last two are `}\n` ? I think it may be useful to catch some sort of errors early.

Reply:
I am going to add a sanity check for starting with `{` and ending with `\n`. I will not match against the two last bytes to avoid having to handle buffering (maybe the last chunk the decoder serves me contains only 1 byte).